dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/gdv/custom/GdvYearlyDecimalTimeseriesDataComponent.kt

```diff
@@ -0,0 +1,121 @@
+package org.dataland.frameworktoolbox.frameworks.gdv.custom
+
+import com.fasterxml.jackson.core.json.JsonWriteFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.addStandardCellWithValueGetterFactory
+import org.dataland.frameworktoolbox.intermediate.components.support.SelectionOption
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.SectionUploadConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.capitalizeEn
+
+/**
+ * A GdvYearlyDecimalTimeseriesDataComponent is an in-memory representation of a generic field
+ * that encodes several values across a span of multi years. It is displayed / upload in a matrix
+ */
+class GdvYearlyDecimalTimeseriesDataComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    /**
+     * TODO
+     */
+    data class TimeseriesRow(val identifier: String, val label: String, val unitSuffix: String)
+
+    var decimalRows: MutableList<TimeseriesRow> = mutableListOf()
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        require(decimalRows.isNotEmpty()) {
+            "Please add at least one decimal row to this GdvYearlyDecimalTimeseriesDataComponent " +
+                "to receive useful output."
+        }
+
+        val fieldDataClass = dataClassBuilder.parentPackage.addClass(
+            name = "${this.identifier.capitalizeEn()}Values",
```

Comment:
A more general question. Do we want to have this suffix "Values" everywhere?

Reply:
I don't mind it. I think it is an explicit indicator that this is not the actual representation of the field but only of part of the field. But I'm open to suggestions :)

Reply:
My angle is the following. Now looking at the backend for GDV there are some files with the suffix Values, the other without. And it is not really clear on the first look, why this is and what it tells the developer.

Reply:
I will resolve this